### PR TITLE
Clarify difference between lsp and hie-bios configuration

### PR DESCRIPTION
**Pull request by lukel97**
Do this by renaming explicit configuration to project configuration

### CODE REVIEW

**Review by fendor — APPROVED** [review]
Can we also explain that we prefer cabal projects over stack projects at the moment? Or should we rather not?

**Comment by lukel97** [comment]
@fendor in what ways do we we prefer cabal projects?

**Comment by fendor** [comment]
if there is a `cabal.project` and a `stack.yaml` we select a cabal project.

**Comment by jneira** [comment]
> if there is a `cabal.project` and a `stack.yaml` we select a cabal project.

for completeness, if cabal is available

i think hie-bios would choose plain ghc under some conditions, right?

**Comment by lukel97** [comment]
@fendor but if there is no `cabal.project` and a `stack.yaml`, then cabal-helper selects a stack project! And a `cabal.project` is not necessary for cabal v2 builds. I'll add a table documenting this

@jneira 
> i think hie-bios would choose plain ghc under some conditions, right?
If neither `cabal` nor `stack` is on the path I presume

**Comment by fendor** [comment]
@jneira no, hie-bios only chooses the cradle based on `hie.yaml`, we are not using the implicit configuration and c-h only chooses between cabal and stack.

@bubba I think with neither *.cabal file nor stack.yaml, we do not find a project context and we are selecting a none-cradle.

File: README.md [hunk 1]
```diff
@@ -305,13 +305,23 @@ There are some settings that can be configured via a `settings.json` file:
...
-that you want to use `stack` then the configuration file would look like:
+HIE will attempt to automatically detect your project configuration and set up
+the environment for GHC. 
+
+| `cabal.project` | `stack.yaml` | `*.cabal` | Project selected |
+|-----------------|--------------|-----------|------------------|
+| ✅              | -            | -         | Cabal v2         |
+| ❌              | ✅           | -         | Stack            |
+| ❌              | ❌           | ✅        | Cabal (v2 or v1) |
+| ❌              | ❌           | ❌        | Plain GHC        |
```
Comment (fendor) [inline README.md]: Currently, I think there is no support for plain ghc projects.

**Review by fendor — CHANGES_REQUESTED** [review]

**Comment by alanz** [comment]
The old behaviour if there was both a `stack.yaml` and `cabal.project` was to look for `dist` or `.stack-work`, and choose based on that if there was only one.  Fallback to whatever.

**Comment by fendor** [comment]
We should add that back, but currently we are definitely using the cabal v2-cradle if there is both.

**Comment by alanz** [comment]
I think the break decision where both are present is arbitrary, there is a hie-bios config file after all, but when it is not present falling back to what the user has at that time does make sense.

**Comment by fendor** [comment]
Should we add this feature after merge or in a consecutive pr?
I think a subsequent pr would be fine.

**Comment by alanz** [comment]
After merge is fine

File: README.md (hunk 1 above)
Comment (lukel97) [reply README.md]: Is this not the direct cradle?

Comment (fendor) [reply README.md]: Afaik, you have to specify the direct cradle yourself via hie.yaml.

Comment (fendor) [reply README.md]: We should maybe document this potential use-case more, though.

**Review by fendor — APPROVED** [review]

**Comment by lukel97** [comment]
@fendor would it be worthwhile to (sometime after the merge) select the direct cradle in these cases?

**Pull request merged**